dom/job/job.go

```diff
@@ -95,13 +82,25 @@ func QueryOne(ctx dao.Context, jobID dao.SUID) (job *Job, has bool, err error) {
 	return
 }
 
-//QueryByPickupDistance is used to select jobs nearest to specified location within specified range limit.
-//NB! This is MVP version, not suitable for production use.
-func QueryByPickupDistance(ctx dao.Context, fromLat float64, fromLon float64, state string, rangeLimit float64, rangeUnit RangeUnit, limit int) ([]*AvailableJob, error) {
-	stmt := jobsInRangeStmt(fromLat, fromLon, state, rangeLimit, rangeUnit, limit)
-	s := dao.NewSQL(stmt)
-	jobs := make([]*AvailableJob, 0, limit)
-	err := s.QueryRows(ctx, func(rows *dao.Rows) error {
+//QueryByPickupDistance is used to select jobs nearest to specified lat/lon within precision in meters.
+// lat/lon - point to count distance from
+// precision - range precision in meters
+// limit - rows limit
+// returns AvailableJob array.
+func QueryByPickupDistance(ctx dao.Context, lat float64, lon float64, state string, precision int32, limit int32) ([]*AvailableJob, error) {
+	distance := fmt.Sprintf("public.st_distance(public.st_point(%v, %v, 4326)::public.geography, public.st_point(pickup_lat, pickup_lon, 4326)::public.geography) as distance", lat, lon)
+	within := fmt.Sprintf("public.st_dwithin(public.st_point(%v, %v, 4326)::public.geography, public.st_point(pickup_lat, pickup_lon, 4326)::public.geography, %v)", lat, lon, precision)
```

Comment:
`st_point(x, y) => st_point(lon, lat)`

> For geodetic coordinates, X is longitude and Y is latitude

https://postgis.net/docs/ST_Point.html

Reply:
yep, got it vice-versa, thanks!